main.tf

```diff
@@ -1051,7 +1051,7 @@ resource "aws_nat_gateway" "this" {
 }
 
 resource "aws_route" "private_nat_gateway" {
-  count = local.create_vpc && var.enable_nat_gateway ? local.nat_gateway_count : 0
+  count = local.create_vpc && var.enable_nat_gateway && var.create_private_nat_gateway_route ? local.nat_gateway_count : 0
```

Comment:
I don't follow the logic here - you want to create a NAT GW but you don't want to route it?

Reply:
@bryantbiggs I can understand why this is hard to follow - 

I do want to route to it, but when running AWS Network Firewall, traffic should generally flow:

private host -> network firewall -> NAT -> IGW

So in order to support that, the 0.0.0.0/0 route on the private route table must actually route to the VPC endpoint ID of the network firewall, not the NAT gateway.

Since this module does not currently support network firewall, the easiest way to accomplish that is to override the creation of the 0.0.0.0/0 NAT route in the module, and create the 0.0.0.0 NFW route outside of it - this way there are no conflicts.

```hcl
module example {
  source = "some-source"
   . . . 
  enable_nat_gateway = true
  create_private_nat_gateway_route = false
}

resource "aws_route" "network_firewall_route" {
  for_each               = toset(concat(module.vpc.private_route_table_ids))
  route_table_id         = each.value
  destination_cidr_block = "0.0.0.0/0"
  vpc_endpoint_id        = module.network_firewall.network_firewall_endpoint_id[0]
}

```

The network firewall's route table would actually have the 0.0.0.0/0 route to the NAT Gateway.

Reply:
@bryantbiggs any update here?